sound/soc/sof/intel/hda.c

```diff
@@ -1128,6 +1137,25 @@ static int hda_sdw_machine_select(struct snd_sof_dev *sdev)
 			else
 				pdata->fw_filename = pdata->desc->default_fw_filename;
 			pdata->tplg_filename = mach->sof_tplg_filename;
+
+			/* see if DMICs are pin-muxed with SoundWire link 2 and 3 */
+			if (!(mach->link_mask & GENMASK(3, 2))) {
```

Comment:
Could there be an exception someday? For example [.link_mask = BIT(2),](https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/intel/common/soc-acpi-intel-cnl-match.c#L57), but it is not a DMIC. There is no DMIC in the case though. Another example is [tgl_sdw_rt711_link1_rt1308_link2_rt715_link0](https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/intel/common/soc-acpi-intel-tgl-match.c#L265) where rt715 is on link 0.
The `if (!(mach->link_mask & GENMASK(3, 2)))` test is still valid for both examples, but I am wondering if we can have a better way to test if PCH DMIC is used.